testplan/testing/multitest/base.py

```diff
@@ -467,7 +454,7 @@ def get_test_context(self):
                             testcase_instance, None
                         )
                         if data is not None:
-                            testcase.__xfail__ = {
+                            testcase.__func__.__xfail__ = {
```

Comment:
There was a side effect from report_target, it was turning the bound method into a free function, but here we need to access now the __func__ attribute.